packages/grafana-prometheus/src/querybuilder/parsing.ts

```diff
@@ -323,15 +308,16 @@ function updateFunctionArgs(expr: string, node: SyntaxNode | null, context: Cont
     return;
   }
   switch (node.type.id) {
-    // In case we have an expression we don't know what kind so we have to look at the child as it can be anything.
-    case Expr:
-    // FunctionCallArgs are nested bit weirdly as mentioned so we have to go one deeper in this case.
-    case FunctionCallArgs: {
+    case FunctionCallBody: {
       let child = node.firstChild;
 
       while (child) {
-        const callArgsExprChild = child.getChild(Expr);
-        const binaryExpressionWithinFunctionArgs = callArgsExprChild?.getChild(BinaryExpr);
+        let binaryExpressionWithinFunctionArgs: SyntaxNode | null;
+        if (child?.type.id === BinaryExpr) {
+          binaryExpressionWithinFunctionArgs = child;
+        } else {
+          binaryExpressionWithinFunctionArgs = child?.getChild(BinaryExpr);
```

Comment:
`child` is truthy here:

```suggestion
        if (child.type.id === BinaryExpr) {
          binaryExpressionWithinFunctionArgs = child;
        } else {
          binaryExpressionWithinFunctionArgs = child.getChild(BinaryExpr);
```